### PR TITLE
format dates correctly

### PR DESCRIPTION
**Pull request by benrudolph**
@biyeun @wpride shoudl fix loading dates, loading empty dates and formatting dates

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 288 - 'moment' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 292 - 'moment' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 293 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 329 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 335 - 'moment' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 336 - Missing semicolon. (semi)

**Comment by biyeun** [comment]
hotfix alert!

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 1]
```diff
@@ -283,6 +283,15 @@ SingleSelectEntry.prototype.onPreProcess = function(newValue) {
     }
 };
 
+$.datetimepicker.setDateFormatter({
+    parseDate: function (date, format) {
+        var d = moment(date, format);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: 'moment' is not defined. (no-undef)

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 2]
```diff
@@ -283,6 +283,15 @@ SingleSelectEntry.prototype.onPreProcess = function(newValue) {
...
+        return d.isValid() ? d.toDate() : false;
+    },
+    formatDate: function (date, format) {
+        return moment(date).format(format);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: 'moment' is not defined. (no-undef)

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 3]
```diff
@@ -314,15 +326,20 @@ function DateTimeEntryBase(question, options) {
...
                 self.answer(moment(newDate).format(self.serverFormat));
             }
-        }).val(self.answer() ? moment(self.answer()).format(self.clientFormat) : self.answer());
+        });
     };
 }
 DateTimeEntryBase.prototype = Object.create(EntrySingleAnswer.prototype);
 DateTimeEntryBase.prototype.constructor = EntrySingleAnswer;
+DateTimeEntryBase.prototype.convertServerToClientFormat = function(date) {
+    return moment(date, this.serverFormat).format(this.clientFormat);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: 'moment' is not defined. (no-undef)

**Pull request merged**